samples_checker/samples_checker/check_samples.py

```diff
@@ -7,43 +7,57 @@
 import argparse
 import utils
 
-arg_parser = argparse.ArgumentParser(description='Check submission xls contains all samples that'
-                                                 'exist in the submitted files')
-arg_parser.add_argument('--sample-xml', required=True, dest='samplexml',
-                        help='XML file containing submission samples')
-arg_parser.add_argument('--file-xml', required=True, dest='filexml',
-                        help='XML file containing submitted files')
-arg_parser.add_argument('--file-path', required=True, dest='filepath',
-                        help='Path to the directory in which submitted files can be found')
-
-args = arg_parser.parse_args()
-sample_xml = args.samplexml
-file_xml = args.filexml
-file_path = args.filepath
-
-submission_samples = utils.get_samples_from_xml(sample_xml)
-submitted_files = utils.get_files_from_xml(file_xml)
-
-has_difference = False
-for file_name in submitted_files:
-    file_type = submitted_files.get(file_name, '')
-    difference_submission_xls_submitted_file, difference_submitted_file_submission_xls =\
-        utils.get_sample_difference(submission_samples, file_path + '/' + file_name, file_type)
-
-    if difference_submitted_file_submission_xls:
-        has_difference = True
-        if file_type == 'vcf':
-            print('The submission does not contain the following genotype ids:', file=sys.stderr)
-        else:
-            print('The submission does not contain the following samples:', file=sys.stderr)
-        print(difference_submitted_file_submission_xls, file=sys.stderr)
-
-    if difference_submission_xls_submitted_file:
-        has_difference = True
-        print('The following samples are not found in file '+file_name, file=sys.stderr)
-        print(difference_submission_xls_submitted_file, file=sys.stderr)
-
-if has_difference:
-    quit(1)
-
-print('Samples checking completed!', file=sys.stdout)
+
+def get_sample_diff(file_path, file_xml, sample_xml, submission_type="t2d"):
+    submission_samples = utils.get_samples_from_xml(sample_xml)
+    submitted_files = utils.get_files_from_xml(file_xml)
+    has_difference = False
+    for file_name in submitted_files:
+        file_type = submitted_files.get(file_name, '')
+        difference_submission_xls_submitted_file, difference_submitted_file_submission_xls = \
+            utils.get_sample_difference(submission_samples, file_path + '/' + file_name, file_type)
+
+        if difference_submitted_file_submission_xls:
+            has_difference = True
+            if file_type == 'vcf':
+                if submission_type == "t2d":
+                    print('The submission does not contain the following genotype ids:', file=sys.stderr)
+                if submission_type == "eva":
```

Comment:
if submission_type param is kept (maybe is not needed if we show the same messages for EVA and AMP), and we pass an incorrect value (i.e., EVA instead of eva), no messages will be shown. 

If this is changed to show the same messages for EVA and AMP, as suggested in a previous comment, then the "if" won't be needed and this wouldn't be a problem.